tests/Microsoft.DotNet.XHarness.Apple.Tests/Orchestration/JustRunOrchestratorTests.cs

```diff
@@ -0,0 +1,369 @@
+        _appInstaller.Reset();
+        _appUninstaller.Reset();
+
+        _justRunOrchestrator = new(
+            _appInstaller.Object,
+            _appUninstaller.Object,
+            _appRunnerFactory.Object,
+            _deviceFinder.Object,
+            _exitCodeDetector.Object,
+            _exitCodeDetector.Object,
+            _logger.Object,
+            _logs,
+            _mainLog.Object,
+            _errorKnowledgeBase.Object,
+            _diagnosticsData,
+            _helpers.Object);
+    }
+
+    [Fact]
+    public async Task OrchestrateSimulatorJustRunTest()
+    {
+        // Setup
+        var testTarget = new TestTargetOs(TestTarget.Simulator_iOS64, "13.5");
+
+        var envVars = new[] { ("envVar1", "value1"), ("envVar2", "value2") };
+
+        _exitCodeDetector
+            .Setup(x => x.DetectExitCode(_appBundleInformation, It.IsAny<IReadableLog>()))
+            .Returns(100)
+            .Verifiable();
+
+        _appRunner
+            .Setup(x => x.RunApp(
+                _appBundleInformation,
+                testTarget,
+                _simulator.Object,
+                null,
+                TimeSpan.FromMinutes(30),
+                false,
+                It.IsAny<IEnumerable<string>>(),
+                envVars,
+                It.IsAny<CancellationToken>()))
+            .ReturnsAsync(new ProcessExecutionResult
+            {
+                ExitCode = 0,
+                TimedOut = false,
+            })
+            .Verifiable();
+
+        // Act
+        var result = await _justRunOrchestrator.OrchestrateRun(
+            _appBundleInformation,
+            testTarget,
+            null,
+            TimeSpan.FromMinutes(30),
+            TimeSpan.FromMinutes(3),
+            expectedExitCode: 100,
+            includeWirelessDevices: false,
+            resetSimulator: true,
+            enableLldb: false,
+            signalAppEnd: false,
+            envVars,
+            Array.Empty<string>(),
+            new CancellationToken());
+
+        // Verify
+        Assert.Equal(ExitCode.SUCCESS, result);
+
+        _deviceFinder.Verify(
+            x => x.FindDevice(testTarget, null, It.IsAny<ILog>(), false),
+            Times.Once);
+
+        VerifySimulatorReset(false);
+        VerifySimulatorCleanUp(false);
+        VerifyDiagnosticData(testTarget);
+
+        _exitCodeDetector.VerifyAll();
+        _appRunner.VerifyAll();
+        _appInstaller.VerifyNoOtherCalls();
+        _appUninstaller.VerifyNoOtherCalls();
+    }
+
+    [Fact]
+    public async Task OrchestrateDeviceJustRunTest()
+    {
+        // Setup
+        var testTarget = new TestTargetOs(TestTarget.Device_iOS, "14.2");
+
+        _exitCodeDetector
+            .Setup(x => x.DetectExitCode(_appBundleInformation, It.IsAny<IReadableLog>()))
+            .Returns(100)
+            .Verifiable();
+
+        var extraArguments = new[] { "--some arg1", "--some arg2" };
+
+        _appRunner
```

Comment:
Similar to above, the setup, result and verify are same or very similar to OrchestrateSimulatorJustRunTest.

Reply:
That setup differs in 3 parameters out of 9, we vary 6 of them in total. I could create a method for setting this up with these 9 optional params but it would be a lot of code because I would have to do it for each mocked method (we have 2 per AppRunner, 2 per AppTester..)

Not sure it's worth it?

Reply:
Same as our other conversation on this topic.